src/args.c

```diff
@@ -0,0 +1,101 @@
+#include "stdbool.h"
+#include "stdio.h"
+#include "stdlib.h"
+#include "string.h"
+
+#include "helper.h"
+#include "args.h"
+
+#define MAX_ARG_LENGTH 30
+
+static bool
+fill_flag_argument(const char *value, struct Option *opt) {
+  bool argument_used = false;
+  if (opt->required_arg == ARG_NONE && value != NULL) {
+    log_err_final("option '%s' doesn't allow an argument", opt->flag);
+  } else if (opt->required_arg == ARG_REQUIRED && value == NULL) {
+    log_err_final("option '%s' requires an argument", opt->flag);
+  }
+  opt->enabled = true;
+  switch (opt->type) {
+    case OPT_BOOLEAN:
+      break;
+    case OPT_STRING:
+      opt->out.string = value;
+      argument_used = true;
+      break;
+    case OPT_INT:
+      opt->out.integer = atoi(value);
+      argument_used = true;
+      break;
+  }
+  return argument_used;
+}
+
+static void
+long_flag_parse(const char *flag, struct Option *opts[], size_t opts_length) {
```

Comment:
This also doesn't check for the difference between ``--dump-ast`` and ``--dump-ast-NONSENSE``.